context-free.md

```diff
@@ -1,22 +1,37 @@
 # Context Free
 
 ##  Who is this programming language for?
-
+People who want to program art, perhaps for interesting geometric shapes.
 
 ## What is easy to do in this language? Why is it easy?
-
+Recursion. We can easily define shapes and recursively create that shape.
 
 ## What is hard to do in this language? Why is it hard?
-
+Art that involves randomness is hard to do. Version 2 does not support 
+function calls and expressions.
 
 ## How did you learn how to program in this language?
-
+Digging around documentation, I found this really helpful
+[page](https://github.com/MtnViewJohn/context-free/wiki/Shape-Adjustments). 
+Looking at the sample code that came with the program was moderately helpful.
+I got help from Sam Andow on understanding loops. Ultimately, what helped me
+the most was and lots and lots of trial and error, which I found to be a 
+pretty frustrating way to learn a language.
```

Comment:
Totally agree with this.  I found trial and error to be the main way I made progress in working with ContextFree, and I feel that it shouldn't be this way for programming languages.  Code and programming should be very explicit, so trial and error is probably a sign that either the documentation or the language itself could be much improved.  I personally think that the documentation could be greatly improved.  It is also possible that because we were constrained by time, we did not put enough effort into properly learning the language.